src/Formats/StructureToProtobufSchema.cpp

```diff
@@ -0,0 +1,214 @@
+#include <Formats/StructureToProtobufSchema.h>
+#include <Formats/StructureToFormatSchemaUtils.h>
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeMap.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeEnum.h>
+#include <Common/StringUtils/StringUtils.h>
+
+namespace DB
+{
+
+using namespace StructureToFormatSchemaUtils;
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+namespace
+{
+
+const std::unordered_map<TypeIndex, String> protobuf_simple_type_names =
+{
+    {TypeIndex::Int8, "int32"},
+    {TypeIndex::UInt8, "uint32"},
+    {TypeIndex::Int16, "int32"},
+    {TypeIndex::UInt16, "uint32"},
+    {TypeIndex::Int32, "int32"},
+    {TypeIndex::UInt32, "uint32"},
+    {TypeIndex::Int64, "int64"},
+    {TypeIndex::UInt64, "uint64"},
+    {TypeIndex::Int128, "bytes"},
+    {TypeIndex::UInt128, "bytes"},
+    {TypeIndex::Int256, "bytes"},
+    {TypeIndex::UInt256, "bytes"},
+    {TypeIndex::Float32, "float"},
+    {TypeIndex::Float64, "double"},
+    {TypeIndex::Decimal32, "bytes"},
+    {TypeIndex::Decimal64, "bytes"},
```

Comment:
why not `int32` and `int64` like in capnproto?

Reply:
Because we serialize/deserialize Decimal32/64 in a different way. In CapnProto format we serialize Decimal32/64 as their underlying int32/64 values into CapnProto int32/64. In Protobuf if we have int32/64 for Decimal32/64 in the schema, we will cast decimal values to integer values, so we will loose precision. So, I decided that better to use `bytes`  by default for Protobuf format to not loose precision